java/client/src/main/java/glide/ffi/resolvers/NativeUtils.java

```diff
@@ -0,0 +1,133 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.ffi.resolvers;
+
+import java.io.*;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.ProviderNotFoundException;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * A simple library class which helps with loading dynamic libraries stored in the JAR archive.
+ * These libraries usually contain implementation of some methods in native code (using JNI - Java
+ * Native Interface).
+ *
+ * @see <a
+ *     href="http://adamheinrich.com/blog/2012/how-to-load-native-jni-library-from-jar">http://adamheinrich.com/blog/2012/how-to-load-native-jni-library-from-jar</a>
```

Comment:
Most of the contents of this file are taken from here. The code is MIT licensed, which is why I preferred this solution over the NativeLibLoader solution that was copy pasted from stack overflow.